examples/tv-casting-app/APIs.md

```diff
@@ -1,123 +1,155 @@
+
+A Casting client (e.g. a mobile phone app) is expected to be a Matter Commissionable Node and a `CastingPlayer` (i.e. a TV) is expected to be a Matter Commissioner. In the context of the [Matter Video Player architecture](https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/master/src/app_clusters/media/VideoPlayerArchitecture.adoc), a `CastingPlayer` would map to [Casting "Video" Player](https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/master/src/app_clusters/media/VideoPlayerArchitecture.adoc#1-introduction). The `CastingPlayer` is expected to be hosting one or more `Endpoints` (similar to [Content Apps](https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/master/src/app_clusters/media/VideoPlayerArchitecture.adoc#1-introduction) in the Matter Video Player architecture) that support one or more Matter Media `Clusters`.
+
+The steps to start a casting session are:
+
+1. [Initialize](#initialize-the-casting-client) the `CastingClient` using the Matter SDK.
+1. [Discover](#discover-casting-players) `CastingPlayer` devices using Matter Commissioner discovery.
+1. [Connect](#connect-to-a-casting-player) to the `CastingPlayer` to discover available endpoints.
+1. [Select](#select-an-endpoint-on-the-casting-player) an available `Endpoint` hosted by the `CastingPlayer`.
+
+Next, you're ready to:
+
+1. [Issue commands](#issuing-commands) to the `Endpoint`.
+1. [Read](#read-operations) playback state.
+1. [Subscribe](#subscriptions) to playback events.
 
 ## Build and Setup
-The Casting Client is expected to consume the Matter TV Casting library built for its respective platform to consume any of the APIs described in this document. Refer to the tv-casting-app READMEs for Android and iOS to understand how the libraries can be built and consumed.
+The Casting Client is expected to consume the Matter TV Casting library built for its respective platform to consume any of the APIs described in this document. Refer to the tv-casting-app READMEs for [Android](android) and [iOS](darwin) to understand how to build and consume each platform's specicific libraries.
+
+### Initialize the Casting Client
+_{Complete Initialization examples: [Android](android/App/app/src/main/java/com/matter/casting/InitializationExample.java) |  [iOS](darwin/TvCasting/TvCasting/TvCastingApp.swift)}_
 
-## Initialization
-_{Complete Initialization examples: [Android](https://github.com/sharadb-amazon/connectedhomeip/blob/casting/examples/tv-casting-app/android/App/app/src/main/java/com/matter/casting/InitializationExample.java) |  [iOS](https://github.com/sharadb-amazon/connectedhomeip/blob/casting/examples/tv-casting-app/darwin/TvCasting/TvCasting/TvCastingApp.swift)}_
+A Casting Client must first define the following [`DataProvider`](android/App/app/src/main/jni/com/matter/casting/params/DataProvider.java) objects for the the Matter Casting library to use throughout the client's lifecycle:
```

Comment:
Can we also use the word "initialize" to indicate the this also initializes the whole SDK i.e. they are not just setting these DataProvider params.

Reply:
I just committed a change to tweak the wording.